libraries/ms-common/inc/spi.h

```diff
@@ -0,0 +1,106 @@
+// HAL SPI
+
+#pragma once
+
+#include <stdint.h>
+
+#include "gpio.h"
+#include "interrupt.h"
+#include "status.h"
+
+
+typdef void (*spi_callback)(uint8_t spi_x, SPIITSource source, void *context);
+
+// For setting data size (applies to both transmission and reception)
+typedef enum {
+  SPI_DATASIZE_4B = 0,
+  SPI_DATASIZE_5B,
+  SPI_DATASIZE_6B,
+  SPI_DATASIZE_7B,
+  SPI_DATASIZE_8B,
+  SPI_DATASIZE_9B,
+  SPI_DATASIZE_10B,
+  SPI_DATASIZE_11B,
+  SPI_DATASIZE_12B,
+  SPI_DATASIZE_13B,
+  SPI_DATASIZE_14B,
+  SPI_DATASIZE_15B,
+  SPI_DATASIZE_16B,
+} SPIDataSize;
```

Comment:
Should data sizes above 8 bits be allowed? Is this configuration even necessary? Could I just default to a data size of 8?